alphagov/paas-product-pages
# Upgrade the static site generation engine, hash CSS and JS and remove IE8 support

## What

Upgrade the static site generation engine, hash CSS and JS and remove IE8 support 

- [x] stop serving IE8 specific stylesheet
- [x] move from `next-mdx-enhanced` to `next-mdx-remote`
- [x] add hashes to static assets (css&js)
- [x] prevent NextJS scripts from loading

## Why 

`next-mdx-enhanced` is being left behind and move to `next-mdx-remote` is encouraged as the former is not compatible with newer version of Nextjs. We want to be able to upgrade. We're taking this opportunity to simplify the templates and drop IE8 support.

## How to review

- **suggest reviewing by commit**
- view deployed work on https://paas-product-pages-rework.london.cloudapps.digital/
**or**
- clone branch
- run `npm install`
- run the site locally: `npm run local` to check pages load and assets have hashes
- build the site `npm run build` to check production build is fine 
- `cd` into `out` folder and serve the page (`npx serve`)
- check page on `localhost:5000` for any issues




![Screenshot 2021-08-02 at 09 24 33](https://user-images.githubusercontent.com/3758555/127828783-4881dc9c-01cf-4edb-b1a7-f70de9ff6316.png)
